OpenRA.Mods.Common/Warheads/CreateEffectWarhead.cs

```diff
@@ -48,55 +48,26 @@ public class CreateEffectWarhead : Warhead
 
 		static readonly BitSet<TargetableType> TargetTypeAir = new BitSet<TargetableType>("Air");
 
-		public ImpactType GetImpactType(World world, CPos cell, WPos pos, Actor firedBy)
+		/// <summary>Checks if there are any actors at impact position and if the warhead is valid against any of them.</summary>
+		ImpactActorType ActorTypeAtImpact(World world, WPos pos, Actor firedBy)
 		{
-			// Matching target actor
-			if (ImpactActors)
-			{
-				var targetType = GetDirectHitTargetType(world, cell, pos, firedBy, true);
-				if (targetType == ImpactTargetType.ValidActor)
-					return ImpactType.TargetHit;
-				if (targetType == ImpactTargetType.InvalidActor)
-					return ImpactType.None;
-			}
+			var anyInvalidActor = false;
 
-			var dat = world.Map.DistanceAboveTerrain(pos);
-			if (dat > AirThreshold)
-				return ImpactType.Air;
-
-			return ImpactType.Ground;
-		}
-
-		public ImpactTargetType GetDirectHitTargetType(World world, CPos cell, WPos pos, Actor firedBy, bool checkTargetValidity = false)
-		{
-			var victims = world.FindActorsOnCircle(pos, WDist.Zero);
-			var invalidHit = false;
-
-			foreach (var victim in victims)
+			// Check whether the impact position overlaps with an actor's hitshape
+			var potentialVictims = world.FindActorsOnCircle(pos, WDist.Zero);
+			foreach (var victim in potentialVictims)
 			{
-				if (!AffectsParent && victim == firedBy)
```

Comment:
Is dropping this intentional? Seems like we don't use `AffectsParent` in this file at all any more.

Reply:
I honestly don't remember, but it probably was since `AffectsParent` defaulting to `false` has always been a horrible source of sublime bugs for this warhead type, as `false` almost never makes sense for audiovisual effects.

I'll restore this, but make this warhead override the default instead.